crates/c-api/src/error.rs

```diff
@@ -14,6 +14,21 @@ impl From<Error> for wasmtime_error_t {
     }
 }
 
+impl Into<Error> for wasmtime_error_t {
+    fn into(self) -> Error {
+        self.error
+    }
+}
+
+#[no_mangle]
+pub extern "C" fn wasmtime_error_new(
+    msg: *const std::ffi::c_char,
+) -> Option<Box<wasmtime_error_t>> {
+    Some(Box::new(wasmtime_error_t::from(anyhow!("{:?}", unsafe {
+        std::ffi::CStr::from_ptr(msg)
+    }))))
+}
```

Comment:
I think instead of `:?` this should use `.to_str()` on the pointer received to avoid Rust-level string escaping which shows up in the debug representation.

Reply:
`.to_str()` leaves me with the potential to get a `Utf8Error` instead of a `&str`. I was not sure what the preferred way to deal with such bad input would be, but based on what I see in `crates/c-api/src/trap.rs`, I think I will go with `String::from_utf8_lossy()`. Sound good?

Reply:
Yeah I think the `_lossy` variant is fine. That's a reality of working with cross-language FFI here but in the context of "just an error message" the lossy variant should be fine.